deps/obs-scripting/obs-scripting-python.c

```diff
@@ -1590,6 +1590,25 @@ bool obs_scripting_python_runtime_linked(void)
 	return (bool)RUNTIME_LINK;
 }
 
+char *obs_scripting_python_version(void)
+{
+	struct dstr ver = {0};
```

Comment:
What you maybe want to do here is initialise `char ver_str[8];` in the calling context (`showPythonVersion`) and pass the address and size of that char array into this function, then use `snprintf` to fill it with the appropriate values (similar to how we "fill" the `python_version_t` with the appropriate values in the first place).

Because you write directly into the char array passed by reference, you don't need to return anything, so the function signature would be `void obs_scripting_python_version(char * version, size_t version_length)`.

I'm also a bit confused that this code works, because you free the `dstr` before returning its internal char pointer, which you'd usually have to do _after_ you passed it via `args` to `QString`.

Reply:
Yeah I wanted to avoid doing that, but it seems to be the most common way we do it, and the cleanest way to do it overall. Changes made.